thoth/report_processing/components/adviser.py

```diff
@@ -142,170 +138,235 @@ def _aggregate_thoth_results_from_ceph(
         """Extract justifications from products in adviser document."""
         if not products:
             _LOGGER.warning(f"No products identified in adviser document: {document_id}")
-            return adviser_dict
+            return justifications_collected
 
         for product in products:
             justifications = product["justification"]
 
             if justifications:
                 # Collect all justifications
                 for justification in justifications:
+
                     if "advisory" in justification:
-                        adviser_dict[document_id] = {
-                            "justification": justification,
-                            "error": True,
-                            "message": justification["advisory"],
-                            "type": "INFO",
-                        }
+                        justifications_collected.append(
+                            {
+                                "document_id": document_id,
+                                "date": datetime_object,
+                                "analyzer_version": analyzer_version,
+                                "justification": justification,
+                                "error": True,
+                                "message": justification["advisory"],
+                                "type": "INFO",
+                            }
+                        )
+                    elif "link" in justification:
+
+                        justifications_collected.append(
+                            {
+                                "document_id": document_id,
+                                "date": datetime_object,
+                                "analyzer_version": analyzer_version,
+                                "justification": justification,
+                                "error": False,
+                                "message": justification["link"],
+                                "type": justification["type"],
+                            }
+                        )
                     else:
-                        adviser_dict[document_id] = {
-                            "justification": justification,
-                            "error": False,
-                            "message": justification["message"],
-                            "type": justification["type"],
-                        }
+                        justifications_collected.append(
+                            {
+                                "document_id": document_id,
+                                "date": datetime_object,
+                                "analyzer_version": analyzer_version,
+                                "justification": justification,
+                                "error": False,
+                                "message": justification["message"],
+                                "type": justification["type"],
+                            }
+                        )
```

Comment:
I guess that in `if` scopes only specific attributes should be set
```
if condition:
    error = ...
    message = ...
    type = ...
...
justifications_collected.append(
                            {
                                "document_id": document_id,
                                "date": datetime_object,
                                "analyzer_version": analyzer_version,
                                "justification": justification,
                                "error": error,
                                "message": message,
                                "type": type,
                            }
```
so that the unnecessary ones won't repeat (doc id, date, analyzer version, justification)